pkg/client/unversioned/helper.go

```diff
@@ -290,7 +291,8 @@ func SetKubernetesDefaults(config *Config) error {
 	if len(config.UserAgent) == 0 {
 		config.UserAgent = DefaultKubernetesUserAgent()
 	}
-	if len(config.Version) == 0 {
+	if len(config.Version) == 0 || apiutil.GetGroup(config.Version) != "" {
```

Comment:
nit: always do len(string) == 0, we don't compare to "" by convention.

Larger problem - this method is supposed to SetKubernetesDefaults() (to the defaults implicit to Kube).  It's certainly not the case now that version has one default across all Kubernetes groups.  So it's likely that this method needs to be split into one that sets defaults per group.  We shouldn't change the existing method, but add new methods SetKubernetesAPIDefaults() and SetKubernetesExperimentalDefaults that invokes this method and defaults the version to the appropriate version per group.